wheel/archive.py

```diff
@@ -54,8 +55,8 @@ def writefile(path, date_time):
         zinfo = zipfile.ZipInfo(path, date_time)
         zinfo.external_attr = st.st_mode << 16
         zinfo.compress_type = zipfile.ZIP_DEFLATED
-        with open(path, 'rb') as fp:
-            zip.writestr(zinfo, fp.read())
+        zip.write(path, arcname=zinfo.filename,
+                  compress_type=zipfile.ZIP_DEFLATED)
```

Comment:
You are ignoring `zinfo.external_attr` here. Are you sure that this does not cause problems?
Also, `compression` is already set to `zipfile.DEFLATED` on the zipfile level so why are you explicitly passing it here?